hudi-client/src/test/java/org/apache/hudi/HoodieClientTestHarness.java

```diff
@@ -0,0 +1,206 @@
+import org.apache.hadoop.hdfs.DistributedFileSystem;
+import org.apache.hadoop.hdfs.MiniDFSCluster;
+import org.apache.hudi.common.HoodieClientTestUtils;
+import org.apache.hudi.common.HoodieTestDataGenerator;
+import org.apache.hudi.common.minicluster.HdfsTestService;
+import org.apache.hudi.common.model.HoodieTableType;
+import org.apache.hudi.common.model.HoodieTestUtils;
+import org.apache.hudi.common.util.FSUtils;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.sql.SQLContext;
+import org.junit.rules.TemporaryFolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The test harness for resource initialization and cleanup.
+ */
+public abstract class HoodieClientTestHarness implements Serializable {
+
+  private static final Logger logger = LoggerFactory.getLogger(HoodieClientTestHarness.class);
+
+  protected transient JavaSparkContext jsc = null;
+  protected transient SQLContext sqlContext;
+  protected transient FileSystem fs;
+  protected String basePath = null;
+  protected TemporaryFolder folder = null;
+  protected transient HoodieTestDataGenerator dataGen = null;
+  protected transient ExecutorService executorService;
+
+  //dfs
+  protected String dfsBasePath;
+  protected transient HdfsTestService hdfsTestService;
+  protected transient MiniDFSCluster dfsCluster;
+  protected transient DistributedFileSystem dfs;
+
+  protected void initSparkContexts(String appName) {
+    // Initialize a local spark env
+    jsc = new JavaSparkContext(HoodieClientTestUtils.getSparkConfForTest(appName));
+    jsc.setLogLevel("ERROR");
+
+    //SQLContext stuff
+    sqlContext = new SQLContext(jsc);
+  }
+
+  protected void initSparkContexts() {
+    initSparkContexts("TestHoodieClient");
+  }
+
+  protected void cleanupSparkContexts() {
+    if (sqlContext != null) {
+      logger.info("Clearing sql context cache of spark-session used in previous test-case");
+      sqlContext.clearCache();
+      sqlContext = null;
+    }
+
+    if (jsc != null) {
+      logger.info("Closing spark context used in previous test-case");
+      jsc.close();
+      jsc.stop();
+      jsc = null;
+    }
+  }
+
+  protected void initTempFolderAndPath() throws IOException {
+    folder = new TemporaryFolder();
+    folder.create();
+    basePath = folder.getRoot().getAbsolutePath();
+  }
+
+  protected void cleanupTempFolderAndPath() throws IOException {
+    if (basePath != null) {
+      new File(basePath).delete();
+    }
+
+    if (folder != null) {
+      logger.info("Explicitly removing workspace used in previously run test-case");
+      folder.delete();
+    }
+  }
+
+  protected void initFileSystem() {
+    if (basePath == null) {
+      throw new IllegalStateException("The base path has not been initialized.");
+    }
+
+    if (jsc == null) {
+      throw new IllegalStateException("The Spark context has not been initialized.");
+    }
+
+    fs = FSUtils.getFs(basePath, jsc.hadoopConfiguration());
```

Comment:
should we just make a  initFileSystemWithDefaultConfiguration(Configuration) method and call that from both this and the method below?